src/rpc/core_rpc_server.cpp

```diff
@@ -2559,9 +2559,14 @@ namespace cryptonote
 
     res.status = CORE_RPC_STATUS_OK;
     res.service_node_states.reserve(pubkey_info_list.size());
-    for (const auto &pubkey_info : pubkey_info_list)
+    res.sns_as_json = cryptonote::obj_to_json_str(pubkey_info_list);
+
+    for (auto &pubkey_info : pubkey_info_list)
     {
       COMMAND_RPC_GET_SERVICE_NODES::response::entry entry = {};
+
+      entry.as_json = cryptonote::obj_to_json_str(pubkey_info);
```

Comment:
This duplicates information that the recipient is already going to receive in sns_as_json, better to avoid. Particularly since people can remotely request this from a public daemon we should try to minimise the bandwidth required and put the onus on the caller to process the information they need from the entire json response.